src/main/java/com/conveyal/datatools/manager/DataManager.java

```diff
@@ -450,6 +452,27 @@ public static boolean isExtensionEnabled(String extensionName) {
         return hasConfigProperty("extensions." + extensionName) && "true".equals(getExtensionPropertyAsText(extensionName, "enabled"));
     }
 
+    /**
+     * In a test environment allows for overriding a specific config value on the server config object.
+     */
+    public static void overrideConfigProperty(String name, String value) {
+        String parts[] = name.split("\\.");
+        ObjectNode node = (ObjectNode) serverConfig;
+
+        //Loop through the dot separated field names to obtain final node and override that node's value.
+        for (int i = 0; i < parts.length; i++) {
+            if (i < parts.length - 1) {
+                if (!node.has(parts[i])) {
+                    node.set(parts[i], JsonUtil.objectMapper.createObjectNode());
+                }
+                node = (ObjectNode) node.get(parts[i]);
```

Comment:
Is this for recursion in the config object?

Reply:
I was just thinking the same. This method could be rewritten to use recursion instead of this for loop, but its functionality is the same.